### ISSUE TITLE
tmseek - potential memory access bug

### ISSUE THREAD

**Issue by kleszcz** [RandyGaul/tinyheaders#6]
In tinymemfile all offsets are signed integers and there is an ambiguity in tmseek. Assert's body suggests that smseek will add offset to the current bytes_read and checks if greater than zero but smseek overwrites bytes_read with offset.  Consider two scenarios:

1. assert error
```c
fp->bytes_read = 0;
tmseek(fp, MAX_INT); // assert ok, bytes_read + offset >= 0;
                     // assign ok, will set bytes_read to MAX_INT - a valid memory offset
tmseek(fp, 1);       // assert error,  bytes_read + offset < 0;
                     // assign ok, will set bytes-read to 1 - still a valid offset
```
2. assign error
```c
fp->bytes_read = 0;
tmseek(fp, MAX_INT); // assert ok, bytes_read + offset >= 0;
                     // assign ok, will set bytes_read to MAX_INT - a valid memory offset
tmseek(fp, -1);      // assert ok,  bytes_read + offset >= 0;
                     // assign error, will set bytes-read to -1 - not a valid offset
```

**Comment by RandyGaul** [comment]
Good points, thanks for bringing them up. Generally when people embed files directly it is for efficiency reasons, e.g. seek times on disk and file IO sys calls are avoided. So adding in machinery to run-time builds I try to avoid. But it's pretty useful to add in one more assert to check writing passed the end of the buffer.

It would probably also makes sense to have tmseek return 1 or 0, and perform some clamping of the bytes_read value. 0 whenever a clamp occurs to signify user error.

The MAX_INT is interesting. Valid file sizes on 64bit OS's can go over MAX_UINT size, so usually we use size_t to deal with indices into files. For the sake of embedded files however last I checked I couldn't get any bigger than 2GB embedded into an EXE before I started hitting trouble on Windows. So instead I think leaving the bytes_read as an int is more helpful, using the sign bit to check behind zero accesses, than to allow a full unsigned 32bit offset range.
